traffic_ops/traffic_ops_golang/servers.go

```diff
@@ -163,3 +167,262 @@ JOIN status st ON s.status = st.id
+				return
+			}
+			w.WriteHeader(status)
+			fmt.Fprintf(w, "%s", errBytes)
+		}
+
+		var dsList []int
+
+		err := json.NewDecoder(r.Body).Decode(&dsList)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		q := r.URL.Query()
+
+		replaceQueryParameter := q["replace"][0]
+		replace, err := strconv.ParseBool(replaceQueryParameter) //accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False. for replace url parameter documentation
+		if err != nil {
+			handleErr(err, http.StatusBadRequest)
+			return
+		}
+
+		serverPathParameter := params["server"]
+		server, err := strconv.Atoi(serverPathParameter)
+		if err != nil {
+			handleErr(err, http.StatusBadRequest)
+			return
+		}
+
+		assignedDSes, err := assignDeliveryServicesToServer(server, dsList, replace, db)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		resp := struct {
+			Response []int `json:"response"`
+			Alerts
+		}{assignedDSes, CreateAlerts(SuccessLevel, "successfully assigned dses to server")}
+		respBts, err := json.Marshal(resp)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, "%s", respBts)
+	}
+}
+
+type Parameter struct {
+	ConfigFile string
+	Name       string
+	Value      string
+}
+
+func assignDeliveryServicesToServer(server int, dses []int, replace bool, db *sqlx.DB) ([]int, error) {
+	//transaction rollback in this functions requires err to be set to the proper error or nil before returning
+	tx, err := db.Beginx()
+	defer func() {
+		if tx != nil {
+			if err != nil {
+				tx.Rollback()
+				return
+			}
+			tx.Commit()
+		}
+	}()
+
+	DBError := fmt.Errorf("database access error")
+	if err != nil {
+		log.Error.Printf("could not begin transaction: %v\n", err)
+		return nil, DBError
+	}
+
+	if replace {
+		//delete currently assigned dses from server
+		deleteCurrent, err := tx.Prepare("DELETE FROM deliveryservice_server WHERE server = $1")
+		if err != nil {
+			log.Error.Printf("could not prepare deliveryservice_server delete statement: %s\n", err)
+			return nil, DBError
+		}
+		_, err = deleteCurrent.Exec(server)
+		if err != nil {
+			log.Error.Printf("could not delete old deliveryservice_server associations for server: %s\n", err)
+			return nil, DBError
+		}
+	}
+
+	//assign new dses
+	dsPqArray := pq.Array(dses)
+	// The common table expressions (CTEs) used below allow for inserting every deliveryService in dses with the server without a loop
+	// the result of the select (for server = 100, dses = [1,2,3]) used by the insert essentially looks like:
+	//          | server
+	//    ---------------
+	//      1   |   100
+	//      2   |   100
+	//      3   |   100
+	// UNNEST is used to turn the array of ds values into, essentially, * FROM ( VALUES (1),(2),(3) )
+	// this allows for a single insert query instead of a loop over the dses as was done in the perl code.
```

Comment:
perl code will be going away -- no need to mention it in the code that's replacing it..